src/routing_table/mod.rs

```diff
@@ -550,29 +550,33 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
             let (dropped_nodes, _opt_our_pfx) = self.split(shorter_pfx);
             result.extend(dropped_nodes);
         }
-        // Merge if necessary, then add empty sections to satisfy the requirement that for each
-        // `i`, the own prefix with the `i`-th bit flipped must be covered. To do this, split each
-        // such prefix recursively until all its parts are either covered or incompatible with the
-        // existing ones. Insert the incompatible ones to cover the required part of the name
-        // space.
-        if self.our_prefix.is_neighbour(&prefix) || self.our_prefix.is_compatible(&prefix) {
-            self.merge(&prefix);
-            let mut missing_pfxs = (0..self.our_prefix.bit_count())
-                .map(|i| self.our_prefix.with_flipped_bit(i))
-                .collect_vec();
-            while let Some(pfx) = missing_pfxs.pop() {
-                if !pfx.is_covered_by(self.sections.keys()) {
-                    if self.sections.keys().any(|p| pfx.is_compatible(p)) {
-                        missing_pfxs.push(pfx.pushed(true));
-                        missing_pfxs.push(pfx.pushed(false));
-                    } else {
-                        let _ = self.sections.insert(pfx, HashSet::new());
-                    }
+
+        // If it's neither our neighbour nor compatible, we need to merge our own until it is
+        // compatible.
+        let mut our_prefix = self.our_prefix;
+        while !our_prefix.is_neighbour(&prefix) && !our_prefix.is_compatible(&prefix) {
```

Comment:
No, the idea is that we accept whatever the section sends us and consider our own routing table invalid rather than the prefix we received. That's because if we missed some update, the `RoutingTableResponse`s can arrive in any order, so they individually may well look like they don't make sense.

The rollback problem should not occur, because after any change that we processed ourselves, we reset the message ID and wait for 30 seconds before requesting the routing table again.